crates/pagecache/src/ds/pagetable.rs

```diff
@@ -40,24 +28,32 @@ fn split_fanout(i: u64) -> (u64, u64) {
 }
 
 struct Node1<T: Send + 'static> {
-    children: Vec<Atomic<Node2<T>>>,
+    children: [Atomic<Node2<T>>; FANOUT as usize],
 }
 
 struct Node2<T: Send + 'static> {
-    children: Vec<Atomic<T>>,
+    children: [Atomic<T>; FANOUT as usize],
 }
 
-impl<T: Send + 'static> Default for Node1<T> {
-    fn default() -> Node1<T> {
-        let children = rep_no_copy!(Atomic::null(); FANOUT as usize);
-        Node1 { children }
+impl<T: Send + 'static> Node1<T> {
+    fn new() -> Box<Node1<T>> {
+        let mut node: Box<Node1<T>> =
+            unsafe { Box::new(std::mem::uninitialized()) };
```

Comment:
we need to allocate the Node_ directly on the heap since it's 2mb and otherwise it could cause a stack overflow

Reply:
That's exactly the type of unsafe block that [the `box` syntax](https://doc.rust-lang.org/unstable-book/language-features/box-syntax.html) will remove :)